server/util/getAtVersionWithRequirements.js

```diff
@@ -0,0 +1,61 @@
+const { Op } = require('sequelize');
+const { getAtVersions } = require('../models/services/AtService');
+
+const getAtVersionWithRequirements = async (
+  atId,
+  exactAtVersion,
+  minimumAtVersion,
+  transaction
+) => {
+  try {
+    if (exactAtVersion) {
+      return exactAtVersion;
+    }
+
+    if (!minimumAtVersion) {
+      throw new Error(
+        'Either exactAtVersion or minimumAtVersion must be provided'
+      );
+    }
+
+    const isMinimumVersionSupported =
+      await minimumAtVersion.supportedByAutomation;
+    if (isMinimumVersionSupported) {
+      return minimumAtVersion;
+    }
+
+    const matchingAts = await getAtVersions({
+      where: {
+        atId,
+        releasedAt: { [Op.gte]: minimumAtVersion.releasedAt }
+      },
+      pagination: {
+        order: [['releasedAt', 'ASC']]
+      },
+      transaction
+    });
+
+    const supportedAts = await Promise.all(
+      matchingAts.map(async version => {
+        const supportedByAutomation = await version.supportedByAutomation;
```

Comment:
The asynchronous request for the `supportedByAutomation` property is necessary since it is virtual and the normal sequelize queries won't return it. Not sure if there are better options.